src/js/player.js

```diff
@@ -2218,27 +2219,23 @@ class Player extends Component {
    *         The current video source when getting
    */
   src(source) {
-    if (source === undefined) {
```

Comment:
video.js throws the following errors:
```js
player.src([]); // No compatible source was found for this media
player.src({}); // The media could not be loaded, either because the server or network failed or because the format is not supported.
```

It looks like vjs 5 special cased empty arrays to have the custom object and with an empty object it basically tried to set an empty string and had it raise the error from the video element.